tagging/taggers.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+from tagger_interface import TaggerInterface
+from git_helper import GitHelper
+from plumbum.cmd import docker
+
+
+def _get_full_name(short_image_name, owner):
+    return f"{owner}/{short_image_name}:latest"
+
+
+def _get_program_version(short_image_name, owner, program):
```

Comment:
For this I have drafted another convenient approach that is using `with` do run and remove a Docker image in order to be able to execute commands while it's running. It could be more efficient since the commands we are running are not altering the container. Here is an example.

```python
import docker
import logging

logger = logging.getLogger(__name__)

class DockerRunner:
    def __init__(self, image_name, docker_client=docker.from_env(), command="sleep infinity"):
        self.container = None
        self.image_name = image_name
        self.command = command
        self.docker_client = docker_client

    def __enter__(self):
        logger.info(f"Creating container for image {self.image_name} ...")
        self.container = self.docker_client.containers.run(
            image=self.image_name, command=self.command, detach=True,
        )
        logger.info(f"Container {self.container.name} created")
        return self.container

    def __exit__(self, exc_type, exc_value, traceback):
        logger.info(f"Removing container {self.container.name} ...")
        if self.container:
            self.container.remove(force=True)
            logger.info(f"Container {self.container.name} removed")
```

Now we can use it just like this.

```python
def test_echo(text="test"):
    cmd = f"echo {text}"
    logger.info(f"Checking if echo command is working ...")
    with DockerRunner("ubuntu") as cont:
        out = cont.exec_run(cmd)
        # another command on the same running container
    assert out.exit_code == 0, f"Command: {cmd} failed"
    result = out.output.decode("utf-8")
    logger.info(f"Command result: {result}")
    assert text in result, f"Command result not expected: {result}"
```

Reply:
Done